File: app/src/main/java/com/bogdan/kolomiiets/holidayslist/NotificationDetail.java
package com.bogdan.kolomiiets.holidayslist;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class NotificationDetail extends AppCompatActivity implements View.OnClickListener {

    private TextView tvTypeOfCelebration;
    private TextView tvPersonName;
    private TextView tvPhone;
    private TextView tvHowOld;
    private Button btnCall, btnSMS, btnClose;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.notification_detail_layout);
        Intent intent = getIntent();

        tvTypeOfCelebration = (TextView) findViewById(R.id.tvTypeOfCelebration);
        tvTypeOfCelebration.setText(new CelebrationType().getCelebrationType(this, intent.getIntExtra(SQLiteDBHelper.KEY_CELEBRATION_TYPE, 0)));

        tvPersonName = (TextView) findViewById(R.id.tvPersonName);
        tvPersonName.setText(intent.getStringExtra(SQLiteDBHelper.KEY_NAME));

        tvPhone = (TextView) findViewById(R.id.tvPhone);
        tvPhone.setText(intent.getStringExtra(SQLiteDBHelper.KEY_PHONE));

        tvHowOld = (TextView) findViewById(R.id.tvHowOld);
        tvHowOld.setText(String.valueOf(new HowOld().getHowOld(intent.getIntExtra(SQLiteDBHelper.KEY_YEAR, 0),
                                                intent.getIntExtra(SQLiteDBHelper.KEY_MONTH, 0),
                                                intent.getIntExtra(SQLiteDBHelper.KEY_DAY, 0))));

        btnCall = (Button) findViewById(R.id.btnCall);
        btnSMS = (Button) findViewById(R.id.btnSMS);
        btnClose = (Button) findViewById(R.id.btnClose);

        btnCall.setOnClickListener(this);
        btnSMS.setOnClickListener(this);
        btnClose.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btnCall:
                Intent intentCall = new Intent(Intent.ACTION_DIAL);
                intentCall.setData(Uri.parse("tel:" + tvPhone.getText().toString()));
                startActivity(intentCall);
                break;
            case R.id.btnSMS:
                Intent intentSMS = new Intent(Intent.ACTION_SENDTO);
                intentSMS.setData(Uri.parse("sms:" + tvPhone.getText().toString()));
                startActivity(intentSMS);
                break;
            case R.id.btnClose:
                finish();
                break;
        }
    }

    @Override
    protected void onUserLeaveHint() {
        super.onUserLeaveHint();
        finish();
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }


}

File: app/src/main/java/com/bogdan/kolomiiets/birthdayreminder/views/NotificationDetail.kt
package com.bogdan.kolomiiets.birthdayreminder.views

import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageButton
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.isVisible
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.EVENT
import com.bogdan.kolomiiets.birthdayreminder.R
import com.bogdan.kolomiiets.birthdayreminder.models.Event
import com.bogdan.kolomiiets.birthdayreminder.utils.calculateAge
import com.bogdan.kolomiiets.birthdayreminder.utils.convertIntTypeToString

class NotificationDetail: AppCompatActivity(), View.OnClickListener {
    private lateinit var personName: TextView
    private lateinit var typeOfCelebration: TextView
    private lateinit var age: TextView
    private lateinit var callBtn: ImageButton
    private lateinit var smsBtn: ImageButton
    private lateinit var closeBtn: ImageButton
    private lateinit var singleCloseBtn: Button
    private lateinit var buttonsContainer: LinearLayout
    private var event: Event? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_notification_detail)

        //init components
        personName = findViewById(R.id.person_name)
        typeOfCelebration = findViewById(R.id.type_of_celebration)
        age = findViewById(R.id.age)
        buttonsContainer = findViewById(R.id.buttons_container)
        callBtn = findViewById(R.id.call_btn)
        smsBtn = findViewById(R.id.sms_btn)
        closeBtn = findViewById(R.id.close_btn)
        singleCloseBtn = findViewById(R.id.single_close_btn)

        //setup onClickListener for components
        callBtn.setOnClickListener(this)
        smsBtn.setOnClickListener(this)
        closeBtn.setOnClickListener(this)
        singleCloseBtn.setOnClickListener(this)

        event = intent.getParcelableExtra(EVENT)

        event?.let {
            personName.text = it.name
            typeOfCelebration.convertIntTypeToString(it.type)
            age.calculateAge(it.celebration_year, it.celebration_month, it.celebration_day)

            //if event doesn't have phone we show only single Close button
            buttonsContainer.isVisible = it.phone.isNotEmpty()
            singleCloseBtn.isVisible = it.phone.isEmpty()
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.call_btn -> {
                val intentCall = Intent(Intent.ACTION_DIAL)
                intentCall.data = Uri.parse("tel:" + event?.phone)
                startActivity(intentCall)
            }
            R.id.sms_btn -> {
                val intentSMS = Intent(Intent.ACTION_SENDTO)
                intentSMS.data = Uri.parse("sms:" + event?.phone)
                startActivity(intentSMS)
            }
            R.id.close_btn,
            R.id.single_close_btn -> finishActivityAndRemoveTask()
        }
    }

    //home button
    override fun onUserLeaveHint() {
        super.onUserLeaveHint()
        finishActivityAndRemoveTask()
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finishActivityAndRemoveTask()
    }

    private fun finishActivityAndRemoveTask() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            finishAndRemoveTask()
        } else finish()
    }
}